core/src/main/scala/cats/evidence/Is.scala

```diff
@@ -0,0 +1,67 @@
+package cats.evidence
+
+import cats.Id
+
+/**
+ * A value of `A Is B` is proof that the types `A` and `B` are the same. More
+ * powerfully, it asserts that they have the same meaning in all type
+ * contexts. This can be a more powerful assertion than `A =:= B` and is more
+ * easily used in manipulation of types while avoiding (potentially
+ * erroneous) coercions.
+ *
+ * `A Is B` is also known as Leibniz equality.
+ */
+abstract class Is[A, B] {
+  def subst[F[_]](fa: F[A]): F[B]
+
+  @inline final def andThen[C](next: B Is C): A Is C =
+    next.subst[A Is ?](this)
+
+  @inline final def compose[C](prev: C Is A): C Is B =
+    prev andThen this
+
+  @inline final def from: B Is A =
```

Comment:
Just changed it actually!

On Tue, Jul 12, 2016 at 7:06 PM P. Oscar Boykin <notifications@github.com>
wrote:

> In core/src/main/scala/cats/evidence/Is.scala
> <https://github.com/typelevel/cats/pull/1178#discussion_r70538890>:
>
> > + * contexts. This can be a more powerful assertion than `A =:= B` and is more
> > + * easily used in manipulation of types while avoiding (potentially
> > + * erroneous) coercions.
> > + *
> > + * `A Is B` is also known as Leibniz equality.
> > + */
> > +abstract class Is[A, B] {
> > +  def subst[F[_]](fa: F[A]): F[B]
> > +
> > +  @inline final def andThen[C](next: B Is C): A Is C =
> > +    next.subst[A Is ?](this)
> > +
> > +  @inline final def compose[C](prev: C Is A): C Is B =
> > +    prev andThen this
> > +
> > +  @inline final def from: B Is A =
>
> I like flip.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/typelevel/cats/pull/1178/files/75d99c44800c0d152e7a093819b87af371f75003#r70538890>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAEdWW9A5h087h9IzJYCnKtQC0KWGYEks5qVB4CgaJpZM4JEy9z>
> .
>